perfTest/recreateDb.js

```diff
@@ -0,0 +1,14 @@
+const { Pool } = require("pg");
+const { parse } = require("pg-connection-string");
+
+const config = parse(process.env.PERF_DATABASE_URL);
+// don't connect to the provided db, or we can't drop it
+const pgPool = new Pool({ ...config, database: undefined });
+
+async function main() {
+  console.log(`Recreating database ${config.database}`);
+  await pgPool.query(`drop database if exists ${config.database};`);
+  await pgPool.query(`create database ${config.database};`);
```

Comment:
Please escape these values, and do this with a single database connection to avoid race conditions. I think this might work:

```js
const pgClient = await pgPool.connect();
try {
  await pgClient.query(`drop database if exists ${pgClient.escape(config.database)};`);
  await pgClient.query(`create database ${pgClient.escape(config.database)};`);
} finally {
  pgClient.release();
}
```

Reply:
That doesn't work, and I'm not really sure what the point of this is. What's the risk of not escaping, given we're already running `drop database` here? If you do want it escaped, can you let me know how I should go about it? I'm not sure how to do it sorry.

Reply:
See https://github.com/brianc/node-postgres/issues/33#issuecomment-1306277

Reply:
Sorry, turns out it's called `escapeIdentifier` rather than simply `escape`:

https://github.com/brianc/node-postgres/blob/3f6760c62ee2a901d374b5e50c2f025b7d550315/packages/pg/lib/client.js#L408-L411

The point of escaping is to avoid potential SQL injection - dropping a different database isn't the only thing you could do - your attack could be much more subtle and nefarious :smiling_imp: